src/partition/ReceivedPartition.cpp

```diff
@@ -462,6 +485,23 @@ void ReceivedPartition::prepareBoundingBox()
     _bb.scaleBy(_safetyFactor);
     _boundingBoxPrepared = true;
   }
+
+  // Expand by user-defined bounding box in case a direct access is desired
+  if (_allowDirectAccess) {
+    auto &other_bb = _mesh->getBoundingBox();
+    _bb.expandBy(other_bb);
+    // TODO: How to treat the scale by safety factor here?
+    // The default value of 0.5 is for the current mappings
+    // still too large in comparison to what we want to do here.
+    // On the other hand, ignoring it makes the xml configuration
+    // inconsistent and does not provide any configurable option for
+    // 'bad' matching geometries.
+    //    _bb.scaleBy(_safetyFactor);
```

Comment:
IMHO you have 2 options:
* use the safety factor here and warn the user when handling a `<use-mesh />` tag.

> Warning: You requested direct access to the mesh "X" without specifying a safety-factor. The default will be used to enlarge your BB.

> Info: You requested direct access to the mesh "X". The safety-factor will not be applied.

* ignore the safety factor here and throw an error if the user asks for direct access and specifies a safety factor when handling a <use-mesh />` tag.



> Error: You requested direct access to the mesh "X" and specified a "safety-factor". This is not allowed, please adjust your BB accordingly prior to calling setBoundingBoxes.

Reply:
When combining direct access with a mapping, it's unfortunately more complicated. We need to safety factor for the mapping, but we should not use it for the "direct access region". The best solution, in my opinion, is to get rid of the safety factor configuration in the long run, but compute it based on the mapping (beyond this PR).

Reply:
Giving the user the information that it is actually ignored is a good idea though. I will try to raise a warning in this case (not an error) and write some comments in the source code.